query_ranges_to_vnodes.cc

```diff
@@ -108,6 +108,13 @@ void query_ranges_to_vnodes_generator::process_one_range(size_t n, dht::partitio
     }
 
     if (ranges.size() < n) {
-        add_range(get_remainder());
+        auto range = get_remainder();
+        const bool is_empty = range.start().has_value() && range.end().has_value() &&
+            range.start()->value().relation_to_keys() - range.end()->value().relation_to_keys() == 2 &&
+            dht::token::to_int64(range.end()->value().token()) - dht::token::to_int64(range.start()->value().token()) == 1 &&
+            (!range.start()->is_inclusive() || !range.end()->is_inclusive());
```

Comment:
Will this detect an empty range like `[1, 1)`?.

Reply:
I made a comment about this above. This patch only cares about one specific case, and it's now explicitly handled in the loop.